packages/@aws-cdk/aws-elasticloadbalancingv2-targets/lib/alb-target.ts

```diff
@@ -33,13 +34,43 @@ export class AlbArnTarget implements elbv2.INetworkLoadBalancerTarget {
 
 /**
  * A single Application Load Balancer as the target for load balancing.
+ *
+ * @deprecated Use AlbListenerTarget instead.
  */
 export class AlbTarget extends AlbArnTarget {
   /**
+   * Create a new ALB target.
+   *
    * @param alb The application load balancer to load balance to
    * @param port The port on which the target is listening
+   *
+   * @deprecated Use AlbListenerTarget instead. This target does not automatically
```

Comment:
No need to deprecate both the class and constructor; the latter will effectively deprecate the former. Can you move this more descriptive deprecation message to the top class-level?